docs/oauth2.md

```diff
@@ -538,13 +664,94 @@ In the following RAR example
 }
 ```
 
-if RabbitMQ node's `resource_server_id` is equal to `finance`, the plugin will compute the following sets of scopes:
+if RabbitMQ nodes `resource_server_id` is equal to `finance`, the plugin will compute the following sets of scopes:
 
 - `finance.read:primary-*/*/*`
 - `finance.write:primary-*/*/*`
 - `finance.configure:primary-*/*/*`
 - `finance.tag:administrator`
 
+
+
+## Advanced usage {#advanced-usage}
+
+### Use default OAuth 2.0 provider {#use-oauth-provider}
+
+As long you have one and only one OAuth 2.0 provider, you can skip this advanced usage although you can use it.
+
+Under the [basic usage](#configure-issuer) section, you configured the `issuer` url or maybe the `jwks_url` along with the TLS settings if needed. This advanced usage configures everything relative to the OAuth provider into a dedicated configuration.
+
+Here is a basic configuration as seen so far:
+
+```ini
+auth_oauth2.resource_server_id = rabbitmq-prod
+auth_oauth2.scope_prefix = rabbitmq.
+auth_oauth2.issuer = https://prodkeycloak:8080/realm/prod
+auth_oauth2.https.cacertfile = /opts/certs/prodcacert.pem
+```
+
+And the equivalent advanced configuration:
+
+```ini
+auth_oauth2.resource_server_id = rabbitmq-prod
+auth_oauth2.scope_prefix = rabbitmq.
+auth_oauth2.default_oauth_provider = prodkeycloak
+
+auth_oauth2.oauth_providers.prodkeycloak.issuer = https://prodkeycloak:8080/realm/prod
+auth_oauth2.oauth_providers.prodkeycloak.https.cacertfile = /opts/certs/prodcacert.pem
+```
+
+This advanced configuration will be more relevant when users present tokens which have been issued or signed by different OAuth 2.0 providers. However, one can still use it provided `auth_oauth2.default_oauth_provider` is set.
+
+### Multiple Resource Servers {#multiple-resource-servers}
+
+Typically, all users that access a RabbitMQ cluster are registered within the same identity provider. And likewise, all tokens targeting the same RabbitMQ cluster also carry the same *audience*. In other words, all users reference a RabbitMQ cluster with the same resource name which must match the value of the `auth_oauth2.resource_server_id` setting.
+
+However, there are some use-cases where RabbitMQ is accessed by users registered in different Identity Providers or tokens are issued for the same RabbitMQ installation but with different *Audience*(s). For these use-cases, RabbitMQ OAuth 2.0 plugin and the Management plugin can be configured with multiple OAuth 2.0 resources.
+
+Below is the OAuth 2.0 plugin configuration for two resources with the ids, `rabbit_prod` and `rabbit_dev`. Both resources (a.k.a. *audience*) are managed by the same Identity Provider whose issuer is `http//some_idp_url`.
```

Comment:
Instead of "Below" lets just say "The following is the...."
Change "ids," to "ids:"
I wouldn't use aka, I would use the full word "also known as"